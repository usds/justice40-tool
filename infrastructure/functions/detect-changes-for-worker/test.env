REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1