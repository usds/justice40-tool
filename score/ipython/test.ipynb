{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
