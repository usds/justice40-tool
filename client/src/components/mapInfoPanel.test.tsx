import * as React from 'react';
import {render} from '@testing-library/react';
import MapInfoPanel from './mapInfoPanel';
import {LocalizedComponent} from '../test/testHelpers';

describe('simulate app starting up, no click on map', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <MapInfoPanel
          className={'someClassName'}
          featureProperties={undefined}
          selectedFeatureId={undefined}
        />
      </LocalizedComponent>,
  );

  it('should match the snapshot of the MapIntroduction component', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('simulate a click on map', () => {
  const featureProperties = {
    'GEOID10': '350459430003',
    'Total population': 960,
    'GEOID10 (percentile)': 0.5784380914343289,
    'Housing burden (percent) (percentile)': 0.10073235017829946,
    'Total population (percentile)': 0.2985685303608629,
    'Linguistic isolation (percent) (percentile)': 0.9623469180109516,
    'Percent of households in linguistic isolation (percentile)': 0.9230800651740774,
    'Poverty (Less than 200% of federal poverty line) (percentile)': 0.947202643271775,
    'Percentage of individuals age 25 or over with less than high school degree (percentile)': 0.7804232684164424,
    'Unemployed civilians (percent) (percentile)': 0.9873599918675144,
    'Score D (percentile)': 0.9321799276549586,
  };
  const selectedFeatureId = 345;

  const {asFragment} = render(
      <LocalizedComponent>
        <MapInfoPanel
          className={'J40Map-module--mapInfoPanel--8Ap7p'}
          featureProperties={featureProperties}
          selectedFeatureId={selectedFeatureId}
        />
      </LocalizedComponent>,
  );

  it('hould match the snapshot of the MapInfoPanel component', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
