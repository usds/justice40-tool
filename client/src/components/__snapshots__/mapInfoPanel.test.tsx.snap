// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simulate a click on map hould match the snapshot of the MapInfoPanel component 1`] = `<DocumentFragment />`;

exports[`simulate app starting up, no click on map should match the snapshot of the MapIntroduction component 1`] = `
<DocumentFragment>
  <div
    class="someClassName"
  >
    <aside>
      <header
        tabindex="0"
      >
        
      How to use the map:
    
      </header>
      <p
        tabindex="0"
      >
        
      Zoom in
    
        <img
          alt="
      a plus icon indicating that the user can zoom in
    *"
          src="test-file-stub"
          tabindex="0"
        />
        
      , search
    
        <img
          alt="
      a magnifying glass icon informing the user can search
    "
          src="test-file-stub"
          tabindex="0"
        />
        
      , or locate yourself 
    
        <img
          alt="
      a gps icon informing the user can locate
    "
          src="test-file-stub"
          tabindex="0"
        />
        
      and select
    
      to see information about any census tract.
    
      </p>
      <p
        tabindex="0"
      >
        
     Things to know:
    
      </p>
      <p
        tabindex="0"
      >
        
      The tool uses census tracts
    
        <img
          alt="
      a tract icon informing the user of census tracts
    "
          src="test-file-stub"
          tabindex="0"
        />
        
    . Census tracts are a small unit of geography. They generally have populations 
    
        <img
          alt="
      a people icon informing the user of communities
    "
          src="test-file-stub"
          tabindex="0"
        />
        
      of between 1,200 - 8,000 people.
    
      </p>
      <p
        tabindex="0"
      >
        
      Communities that are disadvantaged live in tracts that experience burdens. These tracts are highlighted 
    
        <img
          alt="
      a blue circle icon informing the user of what a disadvantaged community is depicted as
    "
          src="test-file-stub"
          tabindex="0"
        />
        
    on the map.
    
      </p>
      <p
        tabindex="0"
      >
        
      The tool ranks most of the burdens using percentiles
    
        <img
          alt="
      An icon that a bell curve or gaussian distribution
    "
          src="test-file-stub"
          tabindex="0"
        />
        
    . Percentiles show how much burden each tract experiences when compared to other tracts.
    
      </p>
      <p
        tabindex="0"
      >
        
      Thresholds
    
        <img
          alt="
      An icon shows a up arrow with a lower limit implying that it's above a threshold
    "
          src="test-file-stub"
          tabindex="0"
        />
        
    , or cutoffs, are used to determine if communities in a tract are disadvantaged. Certain burdens use percentages 
    
        <img
          alt="
      An icon that depicts a part of pie chart being removed
    "
          src="test-file-stub"
          tabindex="0"
        />
        
      or a simple yes/no
    
        <img
          alt="
      An icon that depicts a check mark conveying yes and/or no
    "
          src="test-file-stub"
          tabindex="0"
        />
        
      .
    
      </p>
      <p
        tabindex="0"
      >
        
      Land within the boundaries of Federally Recognized Tribes and point locations for Alaska Native Villages are highlighted
    
        <img
          alt="
      a blue circle icon informing the user of what a disadvantaged community is depicted as
    "
          src="test-file-stub"
          tabindex="0"
        />
        
      on the map. These communities are also considered disadvantaged.
    
      </p>
    </aside>
  </div>
</DocumentFragment>
`;
